org.uqbar.project.wollok.lib/src/wollok.wlk

```diff
@@ -565,15 +588,12 @@ package lang {
 		}
 		method isPrime() {
 			if (self == 1) return false
-			var _prime = true
-			(2..self - 1).forEach({ i =>
-				// Horrible definition, but return doesn't exit from loop
-				// and I need fold, any methods in Range
-				if(self % i == 0) _prime = false 
-			})
-			return _prime
+			return (2..(self.div(2) + 1)).any({ i => self % i == 0 }).negate()
```

Comment:
Why self.div(2) instead of self / 2?
Or also self.sqrt() or similar.


Also we could replace self % i == 0 by self.isDivisibleBy(i) or i.divides(self)